packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/common/get-transaction-receipt-by-tx-id.ts

```diff
@@ -0,0 +1,203 @@
+    methodName,
+    ...req.params,
+  );
+  const reqTxID = req.params[1];
+  const block: common.Block = BlockDecoder.decode(out);
+  const blockJson = JSON.parse(JSON.stringify(block));
+  log.warn(blockJson);
+  log.warn("block number");
+  log.warn(blockJson.header.number);
+  //TODO: return block number
+  const transactReceipt: GetTransactionReceiptResponse = {};
+  transactReceipt.blockNumber = blockJson.header.number;
+  const txIDs = [];
+  if (block.data != undefined && block.data.data != undefined) {
+    const blockDatas = block.data.data;
+    log.warn("tranversing block datas");
+    for (let i = 0; i < blockDatas.length; i++) {
+      const blockData = JSON.parse(JSON.stringify(blockDatas[i]));
+      log.warn("blockdata tranverse");
+      if (
+        blockData.payload == undefined ||
+        blockData.payload.header == undefined ||
+        blockData.payload.header.channel_header == undefined
+      )
+        continue;
+
+      const payloadChannelHeader = blockData.payload.header.channel_header;
+      if (payloadChannelHeader.tx_id != undefined) {
+        txIDs.push(payloadChannelHeader.tx_id);
+        if (payloadChannelHeader.tx_id != reqTxID) continue;
+        log.warn("printing txid---");
+        log.warn(payloadChannelHeader.tx_id);
+        log.warn(blockData);
+        log.warn("payloadchannelheader");
+        log.warn(payloadChannelHeader);
+        //TODO: get payloadChannelHeader channel_id
+        transactReceipt.channelID = payloadChannelHeader.channel_id;
+        log.warn("payloadchannel data");
+        log.warn(blockData.payload.data);
+        if (
+          blockData.payload.data == undefined ||
+          blockData.payload.data.actions == undefined
+        ) {
+          continue;
+        }
+        const payloadDataActions = blockData.payload.data.actions;
+        log.warn("actions");
+        log.warn(payloadDataActions);
+        if (
+          payloadDataActions[0].header == undefined ||
+          payloadDataActions[0].payload == undefined
+        )
+          continue;
+        const actionsHeader = payloadDataActions[0].header;
+        const actionsPayload = payloadDataActions[0].payload;
+        log.warn("actionsHeader");
+        transactReceipt.transactionCreator = actionsHeader.creator;
+        //TODO: get actionsHeader.creator for transaction creator
+        log.warn(actionsHeader);
+        log.warn("actionsPayload");
+        log.warn(actionsPayload);
+        if (actionsPayload.chaincode_proposal_payload == undefined) continue;
+        const chainCodeProposal = actionsPayload.chaincode_proposal_payload;
+        log.warn("chain code proposal");
+        log.warn(chainCodeProposal);
+        if (
+          chainCodeProposal.input == undefined ||
+          chainCodeProposal.input.chaincode_spec == undefined ||
+          chainCodeProposal.input.chaincode_spec.chaincode_id == undefined
+        )
+          continue;
+        /*const chainCodeID =
```

Comment:
@jscode017 If the commented code is not needed please remove it instead of having it commented unless something specific warrants it to stay this way, like if it's an example snippet and it's part of the JSdocs